File: selectionSort.py
def selectionSort(array):
    for i in range(len(array)):
        minIndex = i
        for j in range(i+1, len(array)):
            if array[j] < array[minIndex]:
                minIndex = j
        array[i], array[minIndex] = array[minIndex], array[i]
    return array


array = [5, 1, 2, 7, 10, 3, 9, 4, 6, 8]
print(selectionSort(array))

File: SelectionSort.java
public class SelectionSort {
    public static void main(String[] args) {
        int[] array = { 5, 2, 6, 1, 3, 7, 4, 9, 8 };
        for (int i = 0; i < array.length - 1; i++) {
            int minIndex = i;
            for (int j = i + 1; j < array.length; j++) {
                if (array[j] < array[minIndex]) {
                    minIndex = j;
                }
            }
            int temp = array[i];
            array[i] = array[minIndex];
            array[minIndex] = temp;
        }
        for (int i = 0; i < array.length; i++) {
            System.out.print(array[i] + " ");
        }
    }
}
